projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js

```diff
@@ -43,42 +38,124 @@ const SubscriptionsPanelPlaceholder = ( { children } ) => {
 	);
 };
 
-function AccessLevelSelectorPanel( {
-	setPostMeta,
+function NewsletterEditorSettingsPanel( {
 	accessLevel,
+	setPostMeta,
 	socialFollowers,
 	emailSubscribers,
 	paidSubscribers,
+	showMisconfigurationWarning,
 } ) {
-	if ( ! showNewsletterAccessPanel() ) {
+	// Only show the panels when the corresponding filter is enabled
+	if ( ! isNewsletterFeatureEnabled() ) {
 		return null;
 	}
 
 	return (
-		<PluginDocumentSettingPanel title={ __( 'Newsletter', 'jetpack' ) }>
-			<NewsletterAccess
-				setPostMeta={ setPostMeta }
+		<PluginDocumentSettingPanel
+			title={ __( 'Newsletter access', 'jetpack' ) }
+			icon={ <JetpackLogo showText={ false } height={ 16 } logoColor="#1E1E1E" /> }
+		>
+			<NewsletterAccessDocumentSettings
 				accessLevel={ accessLevel }
+				setPostMeta={ setPostMeta }
 				socialFollowers={ socialFollowers }
 				emailSubscribers={ emailSubscribers }
 				paidSubscribers={ paidSubscribers }
+				showMisconfigurationWarning={ showMisconfigurationWarning }
 			/>
 		</PluginDocumentSettingPanel>
 	);
 }
 
+function NewsletterPrePublishSettingsPanel( {
+	accessLevel,
+	setPostMeta,
+	socialFollowers,
+	emailSubscribers,
+	paidSubscribers,
+	isModuleActive,
+	showMisconfigurationWarning,
+} ) {
+	const { tracks } = useAnalytics();
+	const { changeStatus, isLoadingModules, isChangingStatus } = useModuleStatus( name );
+
+	const enableSubscriptionsModule = () => {
+		tracks.recordEvent( 'jetpack_editor_subscriptions_enable' );
+		return changeStatus( true );
+	};
+
+	// Subscriptions will not be triggered for a post that was already published in the past.
+	const shouldLoadSubscriptionPlaceholder = useSelect( select => {
+		const meta = select( editorStore ).getEditedPostAttribute( 'meta' );
+		return ! isModuleActive && ! isLoadingModules && ! meta?.jetpack_post_was_ever_published;
+	} );
+
+	return (
+		<PluginPrePublishPanel
```

Comment:
This panel ends up showing, but without anything in it on self-hosted sites where the module is enabled:

<img width="310" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/4ada373e-0221-4a62-aee3-f9cc2d11692c">

Maybe we should hide it altogether unless either the module is off, or the newsletter feature is enabled?

Reply:
Good point. Let me look into it...

Reply:
I prevent displaying any panel if the feature is not enabled.